index.js

```diff
@@ -23,13 +23,34 @@ module.exports.pitch = function (request) {
 		"// Hot Module Replacement",
 		"if(module.hot) {",
 		"	// When the styles change, update the <style> tags",
-		"	if(!content.locals) {",
-		"		module.hot.accept(" + loaderUtils.stringifyRequest(this, "!!" + request) + ", function() {",
-		"			var newContent = require(" + loaderUtils.stringifyRequest(this, "!!" + request) + ");",
-		"			if(typeof newContent === 'string') newContent = [[module.id, newContent, '']];",
+		"	module.hot.accept(" + loaderUtils.stringifyRequest(this, "!!" + request) + ", function() {",
+		"		var newContent = require(" + loaderUtils.stringifyRequest(this, "!!" + request) + ");",
+		"		if(typeof newContent === 'string') newContent = [[module.id, newContent, '']];",
+		"		var localsSame = true;",
```

Comment:
Is the name `locals` available ?

Reply:
I guess. But isn’t that a worse name? `localsSame` answers the question “Are the locals the same?” while `locals` sounds like an object or array of local somethings.

Reply:
The logic checks for identity of the locals (it's in the code), so if the name `locals` isn't already declared, `locals` is shorter and is imho clearer further below 

```js
// Do we have (new) locals => no HMR
if (!locals) throw new Error('..')
```

Reply:
Your comment is wrong.

- Do we have more locals than before? => no HMR
- Do we have fewer locals than before? => no HMR
- Do we have the same number of locals as before, but some changed? => no HMR
- Otherwise HMR

In other words:

- Are the locals the same? HMR
- Otherwise no HMR